toolkit/tools/internal/network/network.go

```diff
@@ -61,3 +64,32 @@ func DownloadFile(url, dst string, caCerts *x509.CertPool, tlsCerts []tls.Certif
 
 	return
 }
+
+// CheckNetworkAccess checks whether the installer environment has network access
+func CheckNetworkAccess() (err error, hasNetworkAccess bool) {
+	const (
+		retryAttempts = 10
+		retryDuration = time.Second
+		squashErrors  = false
+	)
+
+	hasNetworkAccess = true
+
+	err = retry.Run(func() error {
+		err := shell.ExecuteLive(squashErrors, "ping", "-c", "1", "www.microsoft.com")
+		if err != nil {
+			logger.Log.Warnf("No network access yet")
+			hasNetworkAccess = false
+		} else {
+			hasNetworkAccess = true
+		}
+
+		return err
+	}, retryAttempts, retryDuration)
```

Comment:
We can now simplify as well:

```suggestion
	err = retry.Run(func() error {
		err := shell.ExecuteLive(squashErrors, "ping", "-c", "1", "www.microsoft.com")
		hasNetworkAccess = err == nil
		if !hasNetworkAccess  {
			logger.Log.Warnf("No network access yet")
		}

		return err
	}, retryAttempts, retryDuration)
```